From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mahesh Bodapati <mbodapat@xilinx.com>
Date: Mon, 10 Oct 2022 15:07:22 +0530
Subject: [PATCH] binutils: Remaining microblaze gdb changes not in binutils-2.46

Add MICROBLAZE_BREAKPOINT_LE define and microblaze_debug() call in
memory_remove_breakpoint that were not yet upstreamed in binutils-2.46.
The other content from 0002 (configure.host, features/Makefile,
microblaze-linux.xml, microblaze-linux.dat, linux-microblaze-low.cc,
configure.srv) is already present in binutils-2.46 upstream.

Upstream-Status: Pending

Signed-off-by: David Holsgrove <david.holsgrove@petalogix.com>
Signed-off-by: Nathan Rossi <nathan.rossi@petalogix.com>
Signed-off-by: Mahesh Bodapati <mbodapat@xilinx.com>
Signed-off-by: Aayush Misra <aayushm@amd.com>
---
 gdb/microblaze-linux-tdep.c |  7 +++++--
 gdb/microblaze-tdep.h       |  1 +
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/gdb/microblaze-linux-tdep.c b/gdb/microblaze-linux-tdep.c
index f3b026fb68..14176621ba 100644
--- a/gdb/microblaze-linux-tdep.c
+++ b/gdb/microblaze-linux-tdep.c
@@ -58,8 +58,11 @@ microblaze_linux_memory_remove_breakpoint (struct gdbarch *gdbarch,
   /* If our breakpoint is no longer at the address, this means that the
      program modified the code on us, so it is wrong to put back the
      old value.  */
-  if (val == 0 && memcmp (bp, old_contents, bplen) == 0)
-    val = target_write_raw_memory (addr, bp_tgt->shadow_contents, bplen);
+  if (val == 0 && memcmp (bp, old_contents, bplen) == 0)
+  {
+      val = target_write_raw_memory (addr, bp_tgt->shadow_contents, bplen);
+      microblaze_debug ("microblaze_linux_memory_remove_breakpoint writing back to memory at addr 0x%lx\n", addr);
+  }
 
   return val;
 }

diff --git a/gdb/microblaze-tdep.h b/gdb/microblaze-tdep.h
index ef5439854d..7fe4fccea0 100644
--- a/gdb/microblaze-tdep.h
+++ b/gdb/microblaze-tdep.h
@@ -119,5 +119,6 @@ struct microblaze_frame_cache
 /* MICROBLAZE_BREAKPOINT defines the breakpoint that should be used.
    Only used for native debugging.  */
 #define MICROBLAZE_BREAKPOINT {0xba, 0x0c, 0x00, 0x18}
+#define MICROBLAZE_BREAKPOINT_LE {0x18, 0x00, 0x0c, 0xba}
 
 #endif /* GDB_MICROBLAZE_TDEP_H */
-- 
2.34.1
